ui/src/Update.elm

```diff
@@ -135,7 +135,16 @@ update msg ({ filters, settings } as model) =
             model ! []
 
         ToggleBuildDetails id ->
-            if List.member id model.expanded then
-                { model | expanded = List.filter (\i -> i /= id) model.expanded } ! []
-            else
-                { model | expanded = id :: model.expanded } ! []
+            { model
+                | expanded =
+                    case model.expanded of
+                        Just current ->
+                            if id == current then
+                                Nothing
+                            else
+                                model.expanded
+
+                        Nothing ->
+                            Just id
```

Comment:
Note: I went with this because swapping expanded entry was making the UI confusing and jiggling.